packages/esm-service-queues-app/src/hooks/useQueueService.ts

```diff
@@ -0,0 +1,37 @@
+import { type FetchResponse, OpenmrsResource, openmrsFetch, restBaseUrl, useOpenmrsSWR } from '@openmrs/esm-framework';
+import { useSystemSetting } from './useSystemSetting';
+import { useMemo } from 'react';
+import type { Concept } from '../types';
+import useSWRImmutable from 'swr/immutable';
+
+function useQueueServices() {
```

Comment:
I see what you are doing here, but ideally I don't think we want our frontend to have to interpret these backend details.  It would be better if we added REST endpoint to retrieve the supported services as Concepts, or alternatively if you only need to display services that are already available on queues, you could hit the "queue" endpoint, iterate over each, and collect the service configured on each.

Reply:
I have a few questions:
1. How many queues can be there in an implementation (the scale)?
2. Does queues use other concepts than the setMembers of the concept defined in the global property?

In case count of queues > max results returned by the REST endpoint, then we will for sure miss out on some services.

It makes more sense to get all services/ statuses from a single place rather than depending on the queues and then fetching them.
Thanks!

Reply:
1. There are maybe dozens of queues max per implementation.
2. Not sure...

> In case count of queues > max results returned by the REST endpoint, then we will for sure miss out on some services.

Isn't this a problem anyway in other places that requires a full list of queues?

I think getting it from the Queue object makes sense as we already need to fetch it.